assets/js/googlesitekit/data/create-fetch-store.test.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * API function to create fetch store tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { createRegistry } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { createFetchStore } from './create-fetch-store';
+
+const STORE_NAME = 'test/some-data';
+const STORE_PARAMS = {
+	baseName: 'getSomeData',
+	controlCallback: ( params ) => {
+		const { aParam, objParam } = params;
+		return API.get( 'core', 'test', 'some-data', {
+			aParam,
+			objParam,
+		} );
+	},
+	reducerCallback: ( state, response, params ) => {
+		const { aParam } = params;
+		return {
+			...state,
+			data: {
+				...( state.data || {} ),
+				[ aParam ]: response,
+			},
+		};
+	},
+	argsToParams: ( objParam, aParam ) => {
+		invariant( 'object' === typeof objParam, 'objParam is required.' );
+		invariant( 'undefined' !== typeof aParam, 'aParam is required.' );
+		return {
+			objParam,
+			aParam,
+		};
+	},
+};
+
+describe( 'createFetchStore store', () => {
+	let apiFetchSpy;
+	let dispatch;
+	let registry;
+	let select;
+	let storeDefinition;
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createRegistry();
+
+		storeDefinition = createFetchStore( STORE_PARAMS );
+		registry.registerStore( STORE_NAME, storeDefinition );
+		dispatch = registry.dispatch( STORE_NAME );
+		store = registry.stores[ STORE_NAME ].store;
+		select = registry.select( STORE_NAME );
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'actions', () => {
+		it( 'includes the expected actions', () => {
+			const fetchStoreDefinition = createFetchStore( {
+				baseName: 'SaveSomeData',
+				controlCallback: async () => true,
+				reducerCallback: ( state ) => state,
```

Comment:
`reducerCallback` is optional so I would omit it from this test, to use only the minimal required args - otherwise it might look like the presence of this is required to get a `receive` action.